napari/components/_layers_list/tests/test_layers_list.py

```diff
@@ -0,0 +1,376 @@
+from napari.components import LayersList
+from napari.layers import Image
+import numpy as np
+
+
+def test_empty_layers_list():
+    """
+    Test instantiating an empty LayersList object
+    """
+    layers = LayersList()
+
+    assert len(layers) == 0
+
+
+def test_adding_layer():
+    """
+    Test adding a Layer
+    """
+    layers = LayersList()
+    layer = Image(np.random.random((10, 10)))
+    layers.append(layer)
+
+    assert len(layers) == 1
+
+
+def test_removing_layer():
+    """
+    Test removing a Layer
+    """
+    layers = LayersList()
+    layer = Image(np.random.random((10, 10)))
+    layers.append(layer)
+    layers.remove(layer)
+
+    assert len(layers) == 0
+
+
+def test_indexing():
+    """
+    Test indexing into a LayerList
+    """
+    layers = LayersList()
+    layer = Image(np.random.random((10, 10)))
+    layer.name = 'image'
+    layers.append(layer)
+
+    assert layers[0] == layer
+    assert layers['image'] == layer
+
+
+def test_insert():
+    """
+    Test inserting into a LayerList
+    """
+    layers = LayersList()
+    layer_a = Image(np.random.random((10, 10)))
+    layer_a.name = 'image_a'
+    layer_b = Image(np.random.random((15, 15)))
+    layer_b.name = 'image_b'
+    layers.append(layer_a)
+    layers.insert(0, layer_b)
+
+    assert [l for l in layers] == [layer_b, layer_a]
+
+
+def test_get_index():
+    """
+    Test getting indexing from LayerList
+    """
+    layers = LayersList()
+    layer_a = Image(np.random.random((10, 10)))
+    layer_a.name = 'image_a'
+    layer_b = Image(np.random.random((15, 15)))
+    layer_b.name = 'image_b'
+    layers.append(layer_a)
+    layers.append(layer_b)
+
+    assert layers.index(layer_a) == 0
+
+
+def test_reordering():
+    """
+    Test indexing into a LayerList by name
+    """
+    layers = LayersList()
+    layer_a = Image(np.random.random((10, 10)))
+    layer_a.name = 'image_a'
+    layer_b = Image(np.random.random((15, 15)))
+    layer_b.name = 'image_b'
+    layer_c = Image(np.random.random((15, 15)))
+    layer_c.name = 'image_c'
+    layers.append(layer_a)
+    layers.append(layer_b)
+    layers.append(layer_c)
+
+    # Rearrange layers by tuple
+    layers[:] = layers[(1, 0, 2)]
+    assert [l for l in layers] == [layer_b, layer_a, layer_c]
+
+    # Swap layers by name
+    layers['image_b', 'image_c'] = layers['image_c', 'image_b']
+    assert [l for l in layers] == [layer_c, layer_a, layer_b]
```

Comment:
```suggestion
    assert list(layers) == [layer_c, layer_a, layer_b]
```
(and elsewhere as applicable)

Reply:
fixed here and elsewhere